tasks/02-uninstall.yaml

```diff
@@ -0,0 +1,60 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: helm-uninstall
+spec:
+  steps:
+    - name: uninstall-instances
+      args:
+        - uninstall
+        - -n=default
+        - --ignore-not-found
+        - --wait
+        - kuadrant-instances
+      command:
+        - helm
+      env:
+        - name: KUBECONFIG
+          value: $(workspaces.shared-workspace.path)/kubeconfig
+      image: quay.io/rhn_support_azgabur/alpine/k8s:1.31.2
+      imagePullPolicy: IfNotPresent
+    - name: uninstall-operators
+      args:
+        - uninstall
+        - -n=default
+        - --ignore-not-found
+        - --wait
+        - kuadrant-operators
+      command:
+        - helm
+      env:
+        - name: KUBECONFIG
+          value: $(workspaces.shared-workspace.path)/kubeconfig
+      image: quay.io/rhn_support_azgabur/alpine/k8s:1.31.2
+      imagePullPolicy: IfNotPresent
+    - name: delete-cert-manager-namespace
+      args:
+        - delete
+        - --ignore-not-found
+        - ns
+        - cert-manager
+      command:
+        - kubectl
+      env:
+        - name: KUBECONFIG
+          value: $(workspaces.shared-workspace.path)/kubeconfig
+      image: quay.io/rhn_support_azgabur/alpine/k8s:1.31.2
+      imagePullPolicy: IfNotPresent
+    - name: delete-kuadrant-crd
+      args:
+        - kubectl get crd -o name | grep "kuadrant" | xargs --no-run-if-empty kubectl delete
+      command:
+        - /bin/bash
+        - -c
+      env:
+        - name: KUBECONFIG
+          value: $(workspaces.shared-workspace.path)/kubeconfig
+      image: quay.io/rhn_support_azgabur/alpine/k8s:1.31.2
+      imagePullPolicy: IfNotPresent
+  workspaces:
```

Comment:
These namespaces and objects get removed as part of Helm uninstall. I can label them as `do not delete` and we can certainly discuss it, but one reason why it is a good idea is to ensure the Kuadrant objects will get deleted so the CRD removal is not blocked. Another reason is, that install creates these namespaces so they need to be removed before install to prevent errors. Helm maybe supports conditional creation of namespaces, but I am not sure.

Reply:
I was just worried about some leftovers. If Helm removes everything I mentioned in my comment and you can't think of any other potential leftovers then I am perfectly fine. I am against any 'do not delete' labeling. After uninstalling everything Kuadrant related should be gone so the cluster is as close to clean freshly provisioned cluster as possible.